### ISSUE TITLE
[DOC] Search Processor (new feature)

### ISSUE THREAD

**Issue by hdhalter** [opensearch-project/documentation-website#2560]
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Search processor is a chain of processors to take in queries preprocess the query, do information retrieval, and rerank/enhance retrieved results.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
markcoh is the SDM, SDE is froh@ and Stavros is the PM

This was targeted for 2.6, but may go experimental in 2.6 and GA in 2.7

Associated issue: https://github.com/opensearch-project/search-processor/issues/80

**Comment by hdhalter** [comment]
Hi @kolchfa-aws , you are listed on the project scorecard as the Doc lead, so please confirm you will be working on this. As noted above, this may not be ready for 2.6 and may go out in 2.7. Labeling as 2.6 just in case.

**Comment by kolchfa-aws** [comment]
@hdhalter Yes, I will be working on this. It's currently scheduled for 2.7.

**Comment by hdhalter** [comment]
Looks like this has been bumped out to 2.8. I'll update the label and milestone.
